janus-idp/backstage-plugins
# feat(quay-common): add permissions support for quay plugin via new plugin quay-common

**Fixes:**

https://issues.redhat.com/browse/RHIDP-1532

**Description:**

This PR adds support for RBAC permission in quay plugin. `quay.view.read` Permission allows the user view the quay plugin components.

**Screenshots:**
---
Without Permission:

![No_permission_alert](https://github.com/janus-idp/backstage-plugins/assets/9964343/37d10857-5315-4011-95b2-0e6401ad11b0)

With permission:

![Quay_tag_details](https://github.com/janus-idp/backstage-plugins/assets/9964343/6e6d2381-ec66-4399-807b-978baf8a6cbb)

![Quay_repository](https://github.com/janus-idp/backstage-plugins/assets/9964343/5b299172-02ab-4c0e-8f6b-513fdf448ab4)
---
**How to test:**



1. [Configure the Quay plugin](https://github.com/janus-idp/backstage-plugins/tree/main/plugins/quay#configuration) and add Github integration.
2. Create a policy file `rbac.csv` for the permission policies
```
    g, user:default/<YOUR_USERNAME>, role:default/quay-viewer
    
    p, role:default/quay-viewer, quay.view.read, read, allow
    
    p, role:default/quay-viewer, catalog-entity.read, read, allow
    p, role:default/quay-viewer, catalog-entity.create, create, allow


```

3. Configure the RBAC Backend plugin to include the quay permission.
  

  ```permission:
    enabled: true
    rbac:
      policies-csv-file: ../<PATH>/<TO>/<CSV>/rbac-policy.csv
      policyFileReload: true
```
4. Annotate the catalog component to include quay image registry

```
    'quay.io/repository-slug': quayadmin/my-quarkus-app
```
5. Register the component and view the image registry (quay) tab to see the image tags working.
6. Now `deny` the `quay.view.read` permission from the `rbac.csv` file and wait for the application to auto-reload. you should see the `Permission required` alert.

---
**Unit tests:**

```
 PASS  src/components/QuayTagPage/QuayTagPage.test.tsx
  QuayTagPage
    ✓ should render permission alert when user does not have view permission (46 ms)
```

```
 PASS  src/components/QuayRepository/QuayRepository.test.tsx
  QuayRepository
    ✓ should render permission alert when user does not have view permission (59 ms)
```   

cc: @kim-tsao @rohitkrai03 @nickboldt @invincibleJai
